### PR TITLE
Fix fail test in `Dialog` component and `utils`

### LINKED ISSUE #1179: Fix failing test in `Dialog`
### Which package has an issue?

@yamada-ui/react

### Package Version

1.3.3

### Description

In what follows, an error occurs as a result of test.

```bash
 ❯ packages/components/modal/tests/dialog.test.tsx:103:45
    101|     await user.click(cancelDialogButton)
    102|     waitFor(() => {
    103|       expect(screen.findByTestId("Dialog")).toBeNull()
       |                                             ^
    104|     })
    105|     await user.click(openDialogButton)
```

### Link to Reproduction

_No response_

### Steps to Reproduce

1. run 'run test'
2. See error

### Browser

None

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

Yes

### Additional Information

None.

### LINKED ISSUE #1180: Fix fail test in `utils`
### Which package has an issue?

@yamada-ui/react

### Package Version

1.3.3

### Description

In what follows, an error occurs as a result of test.

```bash
AssertionError: expected undefined to be 'success' // Object.is equality

Ignored nodes: comments, script, style
<html>
  <head />
  <body>
    <div />
  </body>
</html>

- Expected: 
"success"

+ Received: 
undefined

 ❯ packages/utils/tests/react.test.tsx:286:38
    284|       })
    285|       await waitFor(async () => {
    286|         expect(result.current.value).toBe("success")
       |                                      ^
    287|       })
    288|     })

```

### Link to Reproduction

_No response_

### Steps to Reproduce

1. run 'pnpm test'
2. See error

### Browser

None

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Do you know how to fix the issue?

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix?

Yes

### Additional Information

None.

### PR DESCRIPTION
**Pull request by setodeve**
Closes #1179 
Closes #1180 

## Description
I fixed fail tests which were after runnning `pnpm test`.

## Is this a breaking change (Yes/No):
No

## Additional Information
No

### CODE REVIEW

**Review by illionillion — APPROVED** [review]

File: packages/components/modal/tests/dialog.test.tsx [hunk 1]
```diff
@@ -91,31 +91,33 @@ describe("<Dialog />", () => {
...
 
     const openDialogButton = await findByTestId("OpenDialog")
 
     await user.click(openDialogButton)
     await expect(findByTestId("Dialog")).resolves.toBeInTheDocument()
     const cancelDialogButton = getByText("dialog-cancel")
     await user.click(cancelDialogButton)
-    waitFor(() => {
-      expect(screen.findByTestId("Dialog")).toBeNull()
+    await waitFor(async () => {
```
Comment (koralle) [inline packages/components/modal/tests/dialog.test.tsx:104]: * `queryBy*` methods are synchronous, so `async` is unnecessary.

Comment (koralle) [inline packages/components/modal/tests/dialog.test.tsx:104]: * `queryBy*` methods are synchronous, so `async` is unnecessary.

File: packages/components/modal/tests/dialog.test.tsx [hunk 2]
```diff
@@ -91,31 +91,33 @@ describe("<Dialog />", () => {
...
+    )
 
     const openDialogButton = await findByTestId("OpenDialog")
 
     await user.click(openDialogButton)
     await expect(findByTestId("Dialog")).resolves.toBeInTheDocument()
     const cancelDialogButton = getByText("dialog-cancel")
     await user.click(cancelDialogButton)
-    waitFor(() => {
-      expect(screen.findByTestId("Dialog")).toBeNull()
```
Comment (koralle) [inline packages/components/modal/tests/dialog.test.tsx:103]: Why you removed `screen`?

File: packages/utils/tests/react.test.tsx [hunk 3]
```diff
@@ -278,12 +278,12 @@ describe("React", () => {
       }
       const { result } = renderHook(() => useAsyncRetry(asyncFunction, []))
       expect(result.current.loading).toBeTruthy()
-      await waitFor(() => {})
-      act(() => {
+
+      act(async () => {
```
Comment (koralle) [inline packages/utils/tests/react.test.tsx:282]: @setodeve 

* `waitFor()` calls `act()` internally, 
* `result.current.retry()` is a synchronous method.

So you should fix this statements:

```ts
act(() => { result.current.retry() })

await waitFor(() => { expect(result.current.value).toBe("success") })
```

**Comment by koralle** [comment]
@setodeve 

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

File: packages/components/modal/tests/dialog.test.tsx (hunk 2 above)
Comment (setodeve) [reply packages/components/modal/tests/dialog.test.tsx:103]: I'm sorry, but I'm returning it because there is no clear reason.

**Pull request merged**